### DISCUSSION TITLE
How to represent AWS instances? And relate it to public IP addresses (maybe private too)

### DISCUSSION THREAD

**Discussion by thiagomarafeli** [netbox-community/netbox#14985] [Help Wanted!] [answered]
I want to put my EC2 instances in NetBox, but I don't feel that Devices and VMs fit exactly in this case.

What's the best way to do it?

And also want to easily realte to a public IP address to it (private would be great too) and an instance type

**Comment by candlerb** [comment, marked as answer]
EC2 instances are, fairly obviously, VMs. The main question is about how you organize them in "clusters" and/or "sites".

One option is to create a single cluster called "AWS", with cluster type "Public Cloud". Another option is to create a separate cluster for each region(\*), e.g. "AWS us-east-1" etc, with cluster type "AWS". It's up to you whether you leave the cluster Site unset, or have a site called "AWS", or a separate Site for each AWS region(\*). If you have multiple AWS accounts, you might want to use Site for account. You can also use Cluster Groups to organize them however you like.

For IP addresses: I suggest you set the public IP address as "primary IP". If you want to document the private IPs then you can create a VRF called "AWS-VPC-\<id\>", create the private IP address within this, and then connect this as the "NAT inside" address to the "primary IP" address object.

Instance types are not yet modelled (but see #5795) and they vary significantly between cloud providers. You could simply add a custom field for instance type if you need it. Otherwise, you can just store CPUs and RAM.

(*) Or availability zone. Just beware that AZ names vary between accounts, e.g. us-east-1a in one account might actually be the same data centre as us-east-1b in a different account, and you have no way of knowing. Amazon hide the AZs this way so they can distribute load between their data centres as they see fit.

**Comment by thiagomarafeli** [comment]
Thanks, @candlerb ! That helped me a lot!

You mentioned the Issue 5795 which says about instance types, but, it seems to me that is stale, unfortunately

I was also curious about the best way to represent a Load Balancer in Netbox... any ideas?

**Reply by candlerb** [reply]
Hmm, I've not attempted to do that. Given that it has an IP address, my first inclination would be to model it as a VM with a specific Role that identifies it as a load balancer (or you could use your instance type custom field)
